core/org.wso2.carbon.core.bootup.validator/src/main/java/org/wso2/carbon/core/bootup/validator/SystemValidator.java

```diff
@@ -219,11 +220,12 @@ private ValidationResult validateKeystoreFingerprint(String certFingerprint) thr
         ValidationResult result = new ValidationResult();
         String msg = null;
         boolean isValid;
-        KeyStore primaryKeyStore = getPrimaryKeyStore();
+        KeyStoreManager keyStoreManager = KeyStoreManager.getInstance(MultitenantConstants.SUPER_TENANT_ID);
```

Comment:
What is the reason to move out from reading the primary key store configured in the file ?
Can the key store configured at the super tenant can be changed at that level ?
If so how that impacts current behaviour

Reply:
`getPrimaryKeyStore()` is a duplicate of the function in the KeystoreManager. Both uses the same keystore configured by the toml file. Therefore it is not needed to read the same file twice.